tests/pykafka/test_simpleconsumer.py

```diff
@@ -55,18 +66,61 @@ def test_offset_resume(self):
         with self._get_simple_consumer(
                 consumer_group='test_offset_resume') as consumer:
             [consumer.consume() for _ in xrange(100)]
-            offsets_committed = self._currently_held_offsets(consumer)
+            offsets_committed = consumer.held_offsets
             consumer.commit_offsets()
 
         with self._get_simple_consumer(
                 consumer_group='test_offset_resume') as consumer:
-            offsets_resumed = self._currently_held_offsets(consumer)
-            self.assertEquals(offsets_resumed, offsets_committed)
+            self.assertEquals(consumer.held_offsets, offsets_committed)
+
+    def test_reset_offset_on_start(self):
+        """Try starting from LATEST and EARLIEST offsets"""
+        with self._get_simple_consumer(
+                auto_offset_reset=OffsetType.EARLIEST,
+                reset_offset_on_start=True) as consumer:
+            earliest = consumer.topic.earliest_available_offsets()
+            earliest_minus_one = consumer.held_offsets
+            self.assertTrue(all(
+                earliest_minus_one[i] == earliest[i].offset[0] - 1
+                for i in earliest.keys()))
+            self.assertIsNotNone(consumer.consume())
 
-    @staticmethod
-    def _currently_held_offsets(consumer):
-        return dict((p.partition.id, p.last_offset_consumed)
-                    for p in consumer._partitions.itervalues())
+        with self._get_simple_consumer(
+                auto_offset_reset=OffsetType.LATEST,
+                reset_offset_on_start=True,
+                consumer_timeout_ms=500) as consumer:
+            latest = consumer.topic.latest_available_offsets()
+            latest_minus_one = consumer.held_offsets
+            self.assertTrue(all(
+                latest_minus_one[i] == latest[i].offset[0] - 1
+                for i in latest.keys()))
+            self.assertIsNone(consumer.consume(block=False))
+
+        difference = sum(latest_minus_one[i] - earliest_minus_one[i]
+                         for i in latest_minus_one.keys())
+        self.assertEqual(difference, self.total_msgs)
+
+    def test_reset_offsets(self):
+        """Test resetting to user-provided offsets"""
+        with self._get_simple_consumer(
+                auto_offset_reset=OffsetType.LATEST,
+                reset_offset_on_start=True) as consumer:
+            part_id, latest_offset = next(  # get a non-empty partition
```

Comment:
Ah yes, thanks. I was being lazy here, not wanting to go through `topic.latest_available_offsets()` but I agree that makes it very unclear what the intent is. I've changed it now, so it reads from latest_available_offsets as it should.